src/meta/src/manager/catalog/fragment.rs

```diff
@@ -207,14 +207,12 @@ where
     pub async fn cancel_create_table_fragments(&self, table_id: &TableId) -> MetaResult<()> {
         let map = &mut self.core.write().await.table_fragments;
 
-        match map.entry(*table_id) {
-            Entry::Occupied(o) => {
-                TableFragments::delete(self.env.meta_store(), &table_id.table_id).await?;
-                o.remove();
-                Ok(())
-            }
-            Entry::Vacant(_) => bail!("table_fragment not exist: id={}", table_id),
+        if let Entry::Occupied(o) = map.entry(*table_id) {
```

Comment:
Is the `vacant` arm reachable? We may add some warning if so.

Reply:
Yes, it might be cleaned when recovery. I will add some warning here.